src/main/java/tc/oc/pgm/match/MatchManagerImpl.java

```diff
@@ -50,16 +73,41 @@ public MatchManagerImpl(Server server, MapLibrary library, MapLoader loader)
     this.server = server;
     this.library = library;
     this.loader = loader;
-    this.matchById = new HashMap<>();
-    this.matchIdByWorldName = new HashMap<>();
     this.count = new AtomicInteger(0);
 
     loadNewMaps();
   }
 
-  @Override
-  public Match createMatch(@Nullable String id, PGMMap map) throws Throwable {
-    if (id == null || id.trim().isEmpty()) id = Integer.toString(count.get());
+  public void createPreMatchAsync(final PGMMap map) {
+    PGM.get().getLogger().fine("Creating pre-match for " + map.getName() + " async");
```

Comment:
This class has its own `logger`

Edit: Not just this line, but several in this class need this change.